src/docker/files/ContainerFilesProvider.ts

```diff
@@ -0,0 +1,232 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See LICENSE.md in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { localize } from '../../localize';
+import { DockerOSType } from '../Common';
+import { DockerApiClient } from '../DockerApiClient';
+import { DirectoryItem, DirectoryItemStat, DockerContainerExecutor, listLinuxContainerDirectory, listWindowsContainerDirectory, statLinuxContainerItem, statWindowsContainerItem } from './ContainerFilesUtils';
+import { DockerUri } from './DockerUri';
+
+class MethodNotImplementedError extends Error {
+    public constructor() {
+        super(localize('docker.files.containerFilesProvider.methodNotImplemented', 'Method not implemented.'));
+    }
+}
+
+class UnrecognizedContainerOSError extends Error {
+    public constructor() {
+        super(localize('docker.files.containerFilesProvider.unrecognizedContainerOS', 'Unrecognized container OS.'));
+    }
+}
+
+export class ContainerFilesProvider extends vscode.Disposable implements vscode.FileSystemProvider {
+    private readonly changeEmitter: vscode.EventEmitter<vscode.FileChangeEvent[]> = new vscode.EventEmitter<vscode.FileChangeEvent[]>();
+
+    public constructor(private readonly dockerClientProvider: () => DockerApiClient) {
+        super(() => {
+            this.changeEmitter.dispose();
+        });
+    }
+
+    public get onDidChangeFile(): vscode.Event<vscode.FileChangeEvent[]> {
+        return this.changeEmitter.event;
+    }
+
+    public watch(uri: vscode.Uri, options: { recursive: boolean; excludes: string[]; }): vscode.Disposable {
+        // As we don't actually support watching files, just return a dummy subscription object...
```

Comment:
Not that I observed; it was more to avoid throwing in cases where the API left a reasonable alternative, just in case VS Code (or some other extension) didn't handle it well.